runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1503,7 +1503,10 @@ TR_J9ServerVM::instanceOfOrCheckCast(J9Class *instanceClass, J9Class* castClass)
             // getComponentFromArrayClass multiple times, or getLeafComponentTypeFromArrayClass.
             // each call requires a remote message, faster and easier to call instanceOfOrCheckCast on the client
             JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
-            stream->write(JITServer::MessageType::VM_instanceOfOrCheckCast, instanceClass, castClass);
+            if (cacheUpdate)
+               stream->write(JITServer::MessageType::VM_instanceOfOrCheckCast, instanceClass, castClass);
+            else
+               stream->write(JITServer::MessageType::VM_instanceOfOrCheckCastNoCacheUpdate, instanceClass, castClass);
```

Comment:
The original implementation has a flaw in that we keep the romMapMonitor while sending a message which can be a lenghty operation. Could you please fix by ending the critical sectin before sending any messages?

Reply:
This should be addressed here: 3148b1f8546548ee5ee0b0291603423f72f3f004. If we end up at this point in this `if` block, we no longer do any remote calls and instead will fall through to the final remote call at the end of this routine